File: samples/apicurio-registry-mcp/src/main/java/io/quarkiverse/langchain4j/sample/apicurio/ImportmapResource.java
package io.quarkiverse.langchain4j.sample.apicurio;

import jakarta.annotation.PostConstruct;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;

import io.mvnpm.importmap.Aggregator;

@ApplicationScoped
@Path("/_importmap")
public class ImportmapResource {

    private String importmap;

    @GET
    @Path("/dynamic.importmap")
    @Produces("application/importmap+json")
    public String importMap() {
        return this.importmap;
    }

    @GET
    @Path("/dynamic-importmap.js")
    @Produces("application/javascript")
    public String importMapJson() {
        return JAVASCRIPT_CODE.formatted(this.importmap);
    }

    @PostConstruct
    void init() {
        Aggregator aggregator = new Aggregator();
        aggregator.addMapping("components/", "/components/");
        this.importmap = aggregator.aggregateAsJson();
    }

    private static final String JAVASCRIPT_CODE = """
            const im = document.createElement('script');
            im.type = 'importmap';
            im.textContent = JSON.stringify(%s);
            document.currentScript.after(im);
            """;
}

File: samples/chatbot-easy-rag-kotlin/src/main/kotlin/io/quarkiverse/langchain4j/sample/chatbot/ImportmapResource.kt
package io.quarkiverse.langchain4j.sample.chatbot

import io.mvnpm.importmap.Aggregator
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces

/**
 * Dynamically create the import map
 */
@ApplicationScoped
@Path("/_importmap")
class ImportmapResource {

    private val importmap = Aggregator(
        mapOf(
            "icons/" to "/icons/",
            "components/" to "/components/",
            "fonts/" to "/fonts/"
        )
    ).aggregateAsJson()

    // See https://github.com/WICG/import-maps/issues/235
    // This does not seem to be supported by browsers yet...
    @GET
    @Path("/dynamic.importmap")
    @Produces("application/importmap+json")
    fun importMap(): String = this.importmap

    @GET
    @Path("/dynamic-importmap.js")
    @Produces("application/javascript")
    fun importMapJson(): String =
        // language=javascript
        """
            const im = document.createElement('script');
            im.type = 'importmap';
            im.textContent = JSON.stringify($importmap);
            document.currentScript.after(im);

            """.trimIndent()
}
